JITM: move jetpack-jitm.js to the JITM package

kbrown9 opened the pull request:
#### Changes proposed in this Pull Request:

The JITM package should be able to be used independently of Jetpack. So, let's move the jetpack-jitm.js file out of Jetpack and into the JITM package.

* Update the monorepo:
    * Build the JITM package in the package.json `build_packages` and `build_production_packages` scripts.
    * Add `jetpack-jitm.js` to the eslint ignore list.
* Update the JITM package files:
    * Move the `jetpack-jitm.js` file to the JITM package.
    * Add a `webpack.config.packages.js` file.
    * Update the `.gitattributes` file.
    * Update the `.gitignore` file.
    * Add the `build-production` and `build-development` scripts to the composer.json file.
    * Add a `package.json` file.
    * Update the location of the jetpack-jitm.js file in `JITM::jitm_enqueue_files`.
    

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:
1. Install and activate this branch and connect to wpcom.
2. Navigate to wp-admin -> Jetpack -> Dashboard. If you haven't already dismissed it, the `Give your site a speed boost` JITM should be displayed. Verify that it looks correct.

#### Proposed changelog entry for your changes:
* n/a - I don't think a changelog entry is needed for these changes.

jeherve commented on projects/packages/jitm/package.json:
```diff
@@ -0,0 +1,33 @@
+		"webpack-cli": "4.5.0"
+	},
+	"engines": {
+		"node": "^12.18.0",
```
Can we bump this to `12.20.1` now that the monorepo requires this version of node (see `.nvmrc`)?

kbrown9 replied:
Fixed in commit 803c759.

jeherve commented on projects/packages/jitm/src/class-jitm.php:
```diff
@@ -115,7 +115,7 @@ public function jitm_enqueue_files() {
 		wp_enqueue_script(
 			'jetpack-jitm-new',
-			Assets::get_file_url_for_environment( '_inc/build/jetpack-jitm.min.js', '_inc/jetpack-jitm.js' ),
+			Assets::get_file_url_for_environment( 'assets/js/jetpack-jitm.min.js', 'assets/js/jetpack-jitm.js', __DIR__ ),
```
In the Lazy Images package, we opted to have the assets inside `src/`, and here we have them in their own `assets/` dir. Would it make sense to move everything inside of `src/` now?

kbrown9 replied:
In commit 803c759, I moved the css files to `src/css` and the js file to `src/js`.

jeherve reviewed (approved):
This looks to me now, and tests well! 🚢

jeherve reviewed (changes requested):
Coming back on my review, sorry. I think we'll need to update https://github.com/Automattic/jetpack/blob/e13487d0dbbe4c743739b6f1a3e37c143a8c2614/projects/plugins/jetpack/tools/builder/sass.js#L259 now that the CSS location has changed.

kbrown9 commented:
> Coming back on my review, sorry. I think we'll need to update
> 
> https://github.com/Automattic/jetpack/blob/e13487d0dbbe4c743739b6f1a3e37c143a8c2614/projects/plugins/jetpack/tools/builder/sass.js#L259
> 
> now that the CSS location has changed.

Good catch! I updated that file in commit f6843ea. I also updated the [file location](https://github.com/Automattic/jetpack/pull/18341/commits/f6843ea8adf1e400104b25c5436c6b970c6f3dc1#diff-cc3950066f06f29e4d2bb142dce3f1c059262a074bbe25757ab33e27cdfd20d9L61) in `projects/plugins/jetpack/.gitattributes`. Please take a look at those changes when you get a chance!

jeherve reviewed (approved):
This looks good to me. 🚢

jeherve commented:
r226390-wpcom
